src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/AmbientValueAttribute.cs

```diff
@@ -22,6 +24,13 @@ public AmbientValueAttribute([DynamicallyAccessedMembers(DynamicallyAccessedMemb
         {
             // The try/catch here is because attributes should never throw exceptions. We would fail to
             // load an otherwise normal class.
+
+            if (!IDesignerHost.IsSupported)
+            {
+                Debug.Assert(!IDesignerHost.IsSupported, "Runtime instantiation of this attribute is not allowed.");
+                return;
+            }
+
             try
             {
                 Value = TypeDescriptor.GetConverter(type).ConvertFromInvariantString(value);
```

Comment:
Was this (and the case for DefaultValueAttribute below) causing an error at runtime or is this to help with linkability (e.g. https://github.com/dotnet/runtime/issues/97842)?

Reply:
Yes, this is to help with #97842 (trimming with the corresponding feature switch is set).